### DISCUSSION TITLE
core_cm55.h compilation correction

### DISCUSSION THREAD

**Discussion by lebihanm** [ARM-software/CMSIS_5#1603] [Q&A] [answered]
Hello,

I tried to compile the core_cm55.h and I found some compilation problem with gcc compiler.

The 2 problems detected are
1. __NVIC_PRIO_BITS not defined if the switch __CHECK_DEVICE_DEFINES is not defined. I would propose to add the declaration out side of the compilation switch, it won't generate any restrictions and could be defined in a seperate configuration file.
2. The type IRQn_TYpe is not defined => it could be declared in the file as
/**
  \brief  int32_t IRQn_Type definition. Add on top of official library.
 */
typedef int32_t IRQn_Type;

Thanks

Regards

**Comment by JonatanAntoni** [comment, marked as answer]
Hi @lebihanm,

the core header files, like `core_cm55.h` for Cortex-M55, are *not* intended to be used directly. The missing defines/types are expected to be provided by an enclosing [device header file](https://arm-software.github.io/CMSIS_5/latest/Core/html/device_h_pg.html).

The device header files are typically shipped in the Device Family Pack (DFP) for your device. If you are using generic devices (no vendor specific peripherals) you can use the [pseudo device header for Cortex-M55](https://github.com/ARM-software/CMSIS_5/blob/develop/Device/ARM/ARMCM55/Include/ARMCM55.h).

Let me know in case you have further questions.

Cheers,
Jonatan

**Reply by lebihanm** [reply]
Hello Jonatan,

Thanks for your reply, it's clear for me.

BR, 

Mathieu
